### DISCUSSION TITLE
OpenTherm for ventilation

### DISCUSSION THREAD

**Discussion by BlazE-1975** [arendst/Tasmota#13036] [Ideas]
Hi

I have a ventilation of the type wolf cwl 300 that speaks opentherm. This device seems to be identical to viessmann vitovent 300 and also to brink excellent 300.
In principle, the connection works via tasmota opentherm but the supported Message IDs are different to the once used in the Tasmota Code.

I found a list with IDs that should be supported:
 ID 2 - Master Configuration
 ID 70 - Status V/H
 ID 71 - Control Setpoint V/H
 ID 72 - Fault Flags / Code V/H
 ID 74 - Configuration/MemberID V(H
 ID 77 - Relative Ventilation
 ID 80 - Supply Inlet Temperature
 ID 82 - Exhaust Inlet Temperature
 ID 89 - TSP setting V/H

Can someone help me moddify the OpenTherm code so it is using these MeassageIDs instead of the used MessageIDs for a boiler?

Thanx and Greatings 
Blaze-1975

**Comment by barbudor** [comment]
My understanding is that you would need to extend the array `OpenThermCommand sns_opentherm_commands[]`
and create the necessary set, parse and telemetry functions

I would assume @yury-sannikov would be the best to help here

**Reply by BlazE-1975** [reply]
I know he is the developer. But how can i contact him. I don't see any contact-links.

**Reply by barbudor** [reply]
I did by tagging him in the message.
Now it's up to him to join the conversation if he wants to.

**Reply by BlazE-1975** [reply]
Thanx for helping

**Comment by yury-sannikov** [comment]
hi @BlazE-1975 
Basically, your task can be divided into 2 big chunks:
- support stats for the new OpenTherm codes
- add new commands to the OT

The first one is relatively easy. Just add whatever you need into the https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_69_opentherm_protocol.ino#L238-L245

Also, I would suggest adding something like 
```
#ifdef OT_WOLF_CWL300
 ... your code
#endif
```
to the code to include the changes only if one needs them and unset the  OT_WOLF_CWL300 by default.

If you need to add specific commands (which I'm not sure you need to), then you have to update the following code https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_69_opentherm.ino#L470-L479
